src/network/networkd-routing-policy-rule.c

```diff
@@ -0,0 +1,394 @@
+                return 0;
+        }
+
+        n = NULL;
+
+        return 0;
+}
+
+int config_parse_routing_policy_rule_fwmark_mask(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_routing_policy_rule_free_ RoutingPolicyRule *n = NULL;
+        _cleanup_free_ char *fwmark = NULL;
+        Network *network = userdata;
+        char *p;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = routing_policy_rule_new_static(network, filename, section_line, &n);
+        if (r < 0)
+                return r;
+
+        fwmark = strdup(rvalue);
+        if (!fwmark)
+                return -ENOMEM;
+
+        p = strchr(fwmark, '/');
+        if (p)
+                *p++ = '\0';
+
+        r = safe_atou32(fwmark, &n->fwmark);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse RPDB rule firewall mark, ignoring: %s", fwmark);
+                return 0;
+        }
+
+        if (p) {
+                r = safe_atou32(p, &n->fwmask);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse RPDB rule mask, ignoring: %s", fwmark);
+                        return 0;
+                }
+        }
+
+        n = NULL;
+
+        return 0;
+}
+
+int config_parse_routing_policy_rule_prefix(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_routing_policy_rule_free_ RoutingPolicyRule *n = NULL;
+        Network *network = userdata;
+        union in_addr_union buffer;
+        const char *prefix, *e;
+        int r, f;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = routing_policy_rule_new_static(network, filename, section_line, &n);
+        if (r < 0)
+                return r;
+
+        /* prefixlen */
+        e = strchr(rvalue, '/');
+        if (e) {
+                unsigned i;
+
+                r = safe_atou(e + 1, &i);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "IP rule prefix length is invalid, ignoring assignment: %s", e + 1);
+                        return 0;
+                }
+
+                n->prefixlen = (unsigned char) i;
+
```

Comment:
you need to check the range for the prefix length, and check that it matches the actual address family used. i.e. specifiyng a 40bit prefix len on ipv6 makes sense, on ipv4 it doesn't.

I figure we should have a parse_ip_prefixlen() call or so, which does this properly and expects a family specifier